src/drivers/github.js

```diff
@@ -702,6 +702,15 @@ class Github {
     return GITHUB_SHA;
   }
 
+  async workflowCheck() {
+    try {
+      // TODO post telemetery merge - only run if in ci?
+      await this.checkWorkflowForTimeoutMinutes();
+    } catch (err) {
+      // noop
+    }
```

Comment:
Why do we need the catch? Can it break by any reason?

Reply:
it could have weird fs or parsing errors, since this is a "nice-ity" if it has any errors it should not stop any normal operation by raising an error.

Reply:
But following the code I can see it pretty nice... reads the workflow files if any and then parse the yaml, looks nice to me. I can see your fear still so... the only thing is that might not belong to the cml api then because it does not returns or sets anything just only warns

Reply:
by "cml api" you mean where this code is living in the project? I decided to do this way so that we could always call `checkworflow` and there is a mechanism to add any other future checks that are forge specific in case we introduce other similar changes which require forge-specific updates. github actions vs gitlab-ci vs bb-pipelines